File: core/npm-package/src/cli/commands/drawErrors.ts
import { Command } from 'commander';
import { Registry } from '../../registry/Registry.js';

export function drawErrorsCommand(program: Command): void {
  program
    .command('drawErrors')
    .description('List all registered errors (hash + message)')
    .action(() => {
      const records = Registry.getInstance().getAll();

      if (records.length === 0) {
        console.log('No errors registered.');
        return;
      }

      const hashWidth = 20;
      console.log(
        'HASH'.padEnd(hashWidth) + 'MESSAGE'
      );

      for (const r of records) {
        console.log(
          r.hash.padEnd(hashWidth) + r.message
        );
      }
    });
}

File: core/pip-library/amend/cli/commands/draw_errors.py
# ── amend drawErrors ─────────────────────────────────────────────────
# Lists all errors: key + decoded message. Nothing else.

from amend.registry.registry import Registry


def draw_errors() -> None:
    records = Registry.get_instance().get_all()

    if not records:
        print('No errors registered.')
        return

    key_width = 55
    print(f"{'KEY':<{key_width}}MESSAGE")

    for r in records:
        msg = r.message
        print(f"{r.key:<{key_width}}{msg}")
